File: Form1.vb
Public Class Form1
    Public RepeatLength As UInt32
    Public CrossfadeStart As UInt32
    Public CrossfadeEnd As UInt32
    Public LoopStart As UInt32
    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        MsgBox("Copyright 2021, 2024 Sylvia Leaf (sylvia-leaf)

This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.

This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.")
    End Sub
    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        MsgBox("1. Please select the repeat region (for instance: 10000 - 99999), and then you can use the 'repeat' function on your audio editor. Repeat only once.
2. The crossfade region will be [repeat end] - [crossfade length] + 1 to [repeat end] + [crossfade length] + 1 (for instance: 80000 - 120000). Make sure that the calculated values are not out of bounds!
3. Save the crossfade region.
4. Crossfade the samples. (This feature might only work for individually selected tracks, depending on the audio editor used.)
5. Load the region to other tracks, then repeat step 4.
6. The new loop region is 30000 - 99999 ([repeat end] + [crossfade length] to [Repeat end].")
    End Sub
    Private Sub NumericUpDown1_ValueChanged(sender As Object, e As EventArgs) Handles NumericUpDown1.ValueChanged
        If NumericUpDown1.Value > NumericUpDown2.Value Then
            NumericUpDown1.Value = NumericUpDown2.Value
        End If
        UpdateRepeatLength()
        UpdateLoopStart()
    End Sub
    Private Sub NumericUpDown2_ValueChanged(sender As Object, e As EventArgs) Handles NumericUpDown2.ValueChanged
        If NumericUpDown2.Value < NumericUpDown1.Value Then
            NumericUpDown2.Value = NumericUpDown1.Value
        End If
        UpdateRepeatLength()
    End Sub
    Sub UpdateRepeatLength()
        RepeatLength = NumericUpDown2.Value - NumericUpDown1.Value
        TextBox1.Text = Trim(Str(RepeatLength)) + " Sample"
        If Not RepeatLength = 1 Then
            TextBox1.Text += "s"
        End If
        TextBox4.Text = Trim(Str(NumericUpDown2.Value)) + " Sample"
        If Not NumericUpDown2.Value = 1 Then
            TextBox4.Text += "s"
        End If
    End Sub
    Private Sub NumericUpDown3_ValueChanged(sender As Object, e As EventArgs) Handles NumericUpDown3.ValueChanged
        If NumericUpDown3.Value > RepeatLength Then
            NumericUpDown3.Value = RepeatLength
        End If
        UpdateCrossfadeLength()
        UpdateLoopStart()
    End Sub
    Sub UpdateCrossfadeLength()
        CrossfadeStart = NumericUpDown2.Value + 1 - NumericUpDown3.Value
        CrossfadeEnd = NumericUpDown2.Value + 1 + NumericUpDown3.Value
        TextBox2.Text = Trim(Str(CrossfadeStart)) + " Sample"
        TextBox3.Text = Trim(Str(CrossfadeEnd)) + " Sample"
        If Not CrossfadeStart = 1 Then
            TextBox2.Text += "s"
        End If
        If Not CrossfadeEnd = 1 Then
            TextBox3.Text += "s"
        End If
    End Sub
    Sub UpdateLoopStart()
        LoopStart = NumericUpDown1.Value + NumericUpDown3.Value
        TextBox5.Text = Trim(Str(LoopStart)) + " Sample"
        If Not LoopStart = 1 Then
            TextBox5.Text += "s"
        End If
    End Sub
End Class

File: Loop Point Calculator 2.0/Form1.cs
using Microsoft.VisualBasic.ApplicationServices;
using Microsoft.VisualBasic.Devices;
using System.ComponentModel;
using System.Reflection.Metadata.Ecma335;
using System.Security.Policy;
using System.Text;
using static System.Runtime.InteropServices.JavaScript.JSType;

namespace Loop_Point_Calculator_2._0
{
    public partial class Form1 : Form
    {
        protected Decimal repeatLength = 0;
        protected Decimal crossfadeStart = 0;
        protected Decimal crossfadeEnd = 0;
        protected Decimal loopStart = 0;

        public Form1()
        {
            InitializeComponent();
        }

        private void button_license_Click(object sender, EventArgs e)
        {
            MessageBox.Show("Copyright 2021, 2024, 2026 Sylvia Leaf (sylvia-leaf)\n " +
                "This program is free software: you can redistribute it and / or modify it under the terms of the " +
                "GNU General Public License as published by the Free Software Foundation, either version 3 of the " +
                "License, or (at your option) any later version.\n " +
                "This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without " +
                "even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU " +
                "General Public License for more details.");
        }

        private void button_help_Click(object sender, EventArgs e)
        {
            MessageBox.Show("1. Please select the repeat region (for instance: 10000 - 99999), and then you can use " +
                "the 'repeat' function on your audio editor. Repeat only once. \n" +
                "2. The crossfade region will be [repeat end] - [crossfade length] + 1 to [repeat end] + [crossfade " +
                "length] + 1 (for instance: 80000 - 120000). Make sure that the calculated values are not out of " +
                "bounds!\n" +
                "3. Save the crossfade region.\n" +
                "4. Crossfade the samples. (This feature might only work for individually selected tracks, " +
                "depending on the audio editor used.)\n" +
                "5. Load the region to other tracks, then repeat step 4.\n" +
                "6. The new loop region is 30000 - 99999 ([repeat end] + [crossfade length] to [Repeat end].");
        }

        private void numericUpDown_repStart_ValueChanged(object sender, EventArgs e)
        {
            numericUpDown_repStart.Value = Math.Min(numericUpDown_repStart.Value, numericUpDown_repEnd.Value);
            updateRepeatLength();
            updateLoopStart();
        }

        private void numericUpDown_repEnd_ValueChanged(object sender, EventArgs e)
        {
            numericUpDown_repEnd.Value = Math.Max(numericUpDown_repStart.Value, numericUpDown_repEnd.Value);
            updateRepeatLength();
        }

        private void numericUpDown_xfLength_ValueChanged(object sender, EventArgs e)
        {
            numericUpDown_xfLength.Value = Math.Min(numericUpDown_xfLength.Value, repeatLength);
            updateCrossfadeLength();
            updateLoopStart();
        }

        protected void updateRepeatLength()
        {
            repeatLength = numericUpDown_repEnd.Value - numericUpDown_repStart.Value;
            textBox_repLength.Text = sampleDisplayPluraliser(repeatLength);
            textBox_loopEnd.Text = sampleDisplayPluraliser(numericUpDown_repEnd.Value);
        }
        protected void updateLoopStart()
        {
            loopStart = numericUpDown_repStart.Value + numericUpDown_xfLength.Value;
            textBox_loopStart.Text = sampleDisplayPluraliser(loopStart);
        }
        protected void updateCrossfadeLength()
        {
            crossfadeStart = numericUpDown_repEnd.Value + 1 - numericUpDown_xfLength.Value;
            crossfadeEnd = numericUpDown_repEnd.Value + 1 + numericUpDown_xfLength.Value;
            textBox_xfStart.Text = sampleDisplayPluraliser(crossfadeStart);
            textBox_xfEnd.Text = sampleDisplayPluraliser(crossfadeEnd);
        }
        protected string sampleDisplayPluraliser(Decimal inputValue)
        {
            if (inputValue == 1)
            {
                return $"{inputValue + " Sample"}";
            } else
            {
                return $"{inputValue + " Samples"}";
            }
        }
    }
}
